### PR TITLE
Flash tool args

### PR DESCRIPTION
**Pull request by godefroi**
This pull request adds a simple feature which allows passing arbitrary arguments to `esptool.py`. This is useful for my specific use-case, which requires me to specify `dio` flash mode (where `qio` is the more common option). Note that while the board option is present for *all* boards (or can be, anyway), it is only passed to `esptool.py` and therefore would be ineffectual for other architectures (i.e. RP2040).

The motivation around this feature was discussed in #592.

### CODE REVIEW

File: interop/src/archconfig.ts [hunk 1]
```diff
@@ -119,6 +119,12 @@ export interface DeviceConfig
      * If pin X (`pins.X`) is to be exposed as Y, then `$pins.Y == X`.
      */
     $pins?: Record<string, string>
+
+    /**
+     * Extra arguments to be passed to the flash tool.
+     * For example, `--flash_mode` for `esptool.py` (ESP32 boards).
+     */
+    flashToolArguments?: string[]
```
Comment (mmoskal) [inline interop/src/archconfig.ts:127]: Please rename to `$flashToolArguments` - the identifiers starting with `$` are not stored in device flash and instead only used by the CLI tooling

**Review by mmoskal — CHANGES_REQUESTED** [review]
Looks good, modulo the `$` !

File: interop/src/archconfig.ts (hunk 1 above)
Comment (godefroi) [reply interop/src/archconfig.ts]: I wondered what the convention there was; with some being specified and others not.

**Comment by godefroi** [comment]
I have changed the implementation from a generic-arguments system to a specific flash-mode implementation. This is because the generic-arguments system wasn't ideal, because specific arguments must appear in specific positions (mixed with the existing arguments) and therefore it wasn't as general as it appeared to be.

**Comment by mmoskal** [comment]
I think most arguments users may want to set, on per-device-type basis, would come right after `write_flash`, no? I would rather have the version from previous commit with flash arguments, so we can also set flash frequency etc.

**Comment by pelikhan** [comment]
Thank you for the PR!

**Pull request closed without merging**